test/e2e/framework/service/util.go

```diff
@@ -52,3 +65,205 @@ func TestReachableHTTPWithRetriableErrorCodes(ctx context.Context, host string,
 		}
 	}
 }
+
+func TestAllEndpointsReachable(ctx context.Context, client clientset.Interface, svc *v1.Service, pod *v1.Pod) error {
```

Comment:
its a large diff so asking here, why do we need an additional function if we are just replacing RC by Deployments?

Reply:
can we not keep using the ones in `TestJig`